mava/utils/checkpointing.py

```diff
@@ -0,0 +1,200 @@
+# Copyright 2022 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import re
+import typing
+import warnings
+from copy import copy
+from datetime import datetime
+from pydoc import locate
+from typing import Any, Dict, Optional, Union
+
+import absl.logging as absl_logging
+import orbax.checkpoint
+from chex import Numeric
+from flax.core.frozen_dict import FrozenDict
+from jax.tree_util import tree_map
+from omegaconf import DictConfig
+
+from mava.types import HiddenStates, LearnerState, Params, RNNLearnerState
+
+CHECKPOINTER_VERSION = 0.1
```

Comment:
Very good to have versioning here! Can you please just add a comment on why.

Reply:
Done!